src/fides/api/service/connectors/saas_connector.py

```diff
@@ -615,86 +633,155 @@ def run_consent_request(
         Return True if 200 OK. Raises a SkippingConsentPropagation exception if no action is taken
         against the service.
         """
+        query_config = self.query_config(node)
+        saas_config = self.saas_config
         logger.info(
             "Starting consent request for node: '{}'",
             node.address.value,
         )
         self.set_privacy_request_state(privacy_request, node, request_task)
-        query_config = self.query_config(node)
+        fired: bool = (
+            False  # True if the SaaS connector was successfully called / completed
+        )
 
-        should_opt_in, filtered_preferences = should_opt_in_to_service(
-            self.configuration.system, privacy_request
+        notice_based_override_fn: RequestOverrideFunction = (
+            SaaSRequestOverrideFactory.get_override(
+                saas_config.type, SaaSRequestType.UPDATE_CONSENT
+            )
         )
 
-        if should_opt_in is None:
-            logger.info(
-                "Skipping consent requests on node {}: No actionable consent preferences to propagate",
-                node.address.value,
+        if notice_based_override_fn:
```

Comment:
here, we're saying if we have registered a notice_based override, we follow the notice-based flow